server/src/main/java/com/linecorp/centraldogma/server/internal/storage/repository/git/GitGcRevision.java

```diff
@@ -0,0 +1,155 @@
+        path = new File(rootDir, "git-gc.revision").toPath();
+        try {
+            channel = FileChannel.open(path,
+                                       StandardOpenOption.CREATE,
+                                       StandardOpenOption.READ,
+                                       StandardOpenOption.WRITE);
+        } catch (IOException e) {
+            throw new StorageException("failed to open the last git revision: " + path, e);
+        }
+
+        boolean success = false;
+        try {
+            final long size;
+            try {
+                size = channel.size();
+            } catch (IOException e) {
+                throw new StorageException("failed to get the file length: " + path, e);
+            }
+
+            if (size == 0) {
+                lastRevision = null;
+            } else if (size == REVISION_LEN) {
+                lastRevision = read();
+            } else {
+                throw new StorageException("incorrect revision length: " + path + " (" + size + " bytes)");
+            }
+            success = true;
+        } finally {
+            if (!success) {
+                close();
+            }
+        }
+    }
+
+    @Nullable
+    Revision lastRevision() {
+        return lastRevision;
+    }
+
+    private Revision read() {
+        final ByteBuffer buf = threadLocalBuffer.get();
+        buf.clear();
+        long pos = 0;
+
+        try {
+            do {
+                final int readBytes = channel.read(buf, pos);
+                if (readBytes < 0) {
+                    throw new EOFException();
+                }
+                pos += readBytes;
+            } while (buf.hasRemaining());
+        } catch (IOException e) {
+            throw new StorageException("failed to read the last git gc revision: " + path, e);
+        }
+
+        buf.flip();
+        return new Revision(buf.getInt());
+    }
+
+    void write(Revision gcRevision) {
+        if (lastRevision != null && lastRevision.major() > gcRevision.major()) {
+            // less than the last gc revision.
+            return;
+        }
+
+        // Build a record.
+        final ByteBuffer buf = threadLocalBuffer.get();
+        buf.clear();
+        buf.putInt(gcRevision.major());
+        buf.flip();
+
+        // Overwrite the gc revision to the file.
+        long pos = 0;
+        try {
+            do {
+                pos += channel.write(buf, pos);
+            } while (buf.hasRemaining());
+        } catch (IOException e) {
+            throw new StorageException("failed to update the last gc revision: " + path, e);
+        }
+
+        if (lastRevision == null ||
+            lastRevision.major() < gcRevision.major()) {
+            lastRevision = gcRevision;
+        }
+    }
+
+    @Override
+    public void close() {
+        try {
+            channel.close();
+        } catch (IOException e) {
+            logger.warn("Failed to close the commit ID database: {}", path, e);
```

Comment:
Failed to close the Git GC revision database?

Reply:
Oops..